### ISSUE TITLE
[animations]FAB doesn't have a ripple effect in container_transition example

### ISSUE THREAD

**Issue by hammer0802** [flutter/flutter#52236]
Hello, Flutter team.
Thank you for your kind support as always.

I found FAB doesn't have a ripple effect in the below file, because it isn't wrapped by `_InkWellOverlay` or just an InkWell Widget.

[source](https://github.com/flutter/packages/blob/master/packages/animations/example/lib/container_transition.dart)

```      floatingActionButton: OpenContainer(
        transitionType: _transitionType,
        openBuilder: (BuildContext context, VoidCallback _) {
          return _DetailsPage();
        },
        closedElevation: 6.0,
        closedShape: const RoundedRectangleBorder(
          borderRadius: BorderRadius.all(
            Radius.circular(_fabDimension / 2),
          ),
        ),
        closedColor: Theme.of(context).colorScheme.secondary,
        closedBuilder: (BuildContext context, VoidCallback openContainer) {
          return SizedBox(
            height: _fabDimension,
            width: _fabDimension,
            child: Center(
              child: Icon(
                Icons.add,
                color: Theme.of(context).colorScheme.onSecondary,
              ),
            ),
          );
        },
      ),
```

As far as I can see this gif, I think FAB should have a ripple effect.

![container_transform_lineup](https://user-images.githubusercontent.com/39510275/76214712-a6966000-6250-11ea-8a90-b4497cc53c8d.gif)
[link](https://github.com/flutter/packages/blob/master/packages/animations/example/demo_gifs/container_transform_lineup.gif)

Is this the intended implementation ?

Thank you,

**Comment by farheen21** [comment]
yes! i guess this will make the animation looks better, @shihaohong if you allow, I would like to work on this .

**Comment by shihaohong** [comment]
Hi @farheen21! The reason that the FAB doesn't have an Ink Ripple is because it doesn't use a [real FAB widget](https://master-api.flutter.dev/flutter/material/FloatingActionButton-class.html) because there are issues with how that works with the transition itself.

It's been a while since I worked on the problem, but I remember one issue being that the elevation shadow was clipping improperly and that I struggled with getting the shape to transition smoothly. 

Please do give it a try! It would be awesome if you figure out a solution to this problem :)

**Comment by farheen21** [comment]
@shihaohong  are talking about the issue #35283. If this is not kindly mention it here i will try to do my best for it.

**Comment by shihaohong** [comment]
@farheen21 that's a separate issue. If you [look at the example app's code](https://github.com/flutter/packages/blob/master/packages/animations/example/lib/container_transition.dart#L239), notice that we aren't using a FloatingActionButton widget, but instead using a `SizedBox` with an `Icon` in it.

**Comment by kdrag0n** [comment]
@shihaohong Is there a specific reason why you need to use the real FAB elevation? I've had good results with the following code:

```dart
    final theme = Theme.of(context);

    return OpenContainer(
      closedElevation: theme.floatingActionButtonTheme.elevation ?? 6,
      closedColor: theme.floatingActionButtonTheme.backgroundColor ?? theme.colorScheme.secondary,
      closedShape: CircleBorder(),
      closedBuilder: (context, openAction) {
        return FloatingActionButton(
          elevation: 0,
          onPressed: openAction,
          tooltip: tooltip,
          shape: Border(),
          child: icon,
        );
      },
      openBuilder: openBuilder,
      tappable: false,
    );
```

This relies on the OpenContainer to simulate the FAB elevation, shape, and fallback background color (to make the transition smoother) while using the real FAB for everything else. It disables the shadow and shape clipping on the real FAB to prevent it from interfering with transitions.

The dynamic FAB elevations (disabled, focus, highlight, hover) are still missing, but the animation looks smooth and the other FAB mechanics are still intact: tooltip, ink splash, dimension, colors, etc. It's not ideal, but I think this is still much better than using SizedBox since it also obeys the current FloatingActionButtonTheme for the visuals of the FAB itself.

**Comment by NekoGryphou** [comment]
Hi, any news about that ?
